### PR TITLE
351 Make FE complient with BE saving logic

### LINKED ISSUE #351: Save RoutePath changes to Postgres DB
BE:
https://gitlab.hsl.fi/jore/jore-map-backend/merge_requests/12

FE:
https://github.com/HSLdevcom/jore-map-ui/pull/366

### PR DESCRIPTION
**Pull request by kronken**
Closes #351  together with https://gitlab.hsl.fi/jore/jore-map-backend/merge_requests/12

### CODE REVIEW

File: src/util/apiClientHelper.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    }
+
+    public static format = (obj: object) => {
+        // Formats the object to include dates as formatted strings, instead of Date objects
+        const entries = Object.entries(obj);
+        const dates = entries
+            .filter(([key, value]: [string, any]) => value instanceof Date)
+            .map(
+                ([key, value]: [string, any]) =>
+                    [key, moment(value).format()] as [string, any]);
```
Comment (neodyymi) [inline src/util/apiClientHelper.ts]: `as [string, string]`

Comment (neodyymi) [reply src/util/apiClientHelper.ts]: As far as I can tell, moment.format() returns a string

File: src/util/apiClientHelper.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        return res;
+    }
+
+    public static format = (obj: object) => {
+        // Formats the object to include dates as formatted strings, instead of Date objects
+        const entries = Object.entries(obj);
+        const dates = entries
+            .filter(([key, value]: [string, any]) => value instanceof Date)
+            .map(
+                ([key, value]: [string, any]) =>
```
Comment (neodyymi) [inline src/util/apiClientHelper.ts]: We just filtered out anything where the value isn't Date, so make this `[string, Date]` ?

File: src/util/ApiClient.tsx [hunk 3]
```diff
@@ -22,13 +29,32 @@ export default class ApiClient {
...
+        const formattedObject = ApiClientHelper.format(object);
         return await fetch(this.getUrl(entityName), {
             method,
-            body: JSON.stringify(object),
+            body: JSON.stringify(formattedObject),
             headers: {
                 Accept: 'application/json',
                 'Content-Type': 'application/json',
             },
+        }).then((response: Response) => {
```
Comment (neodyymi) [inline src/util/ApiClient.tsx]: This could be changed into await too

File: src/util/apiClientHelper.ts (hunk 1 above)
Comment (kronken) [reply src/util/apiClientHelper.ts]: true dat

File: src/util/apiClientHelper.ts (hunk 2 above)
Comment (kronken) [reply src/util/apiClientHelper.ts]: true

File: src/util/ApiClient.tsx (hunk 3 above)
Comment (kronken) [reply src/util/ApiClient.tsx]: true

**Pull request merged**